pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/AuxClassPathController.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+
+package net.sourceforge.pmd.util.fxdesigner;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+import java.util.ResourceBundle;
+
+import org.controlsfx.validation.ValidationSupport;
+
+import net.sourceforge.pmd.PMDConfiguration;
+import net.sourceforge.pmd.util.ClasspathClassLoader;
+import net.sourceforge.pmd.util.fxdesigner.util.DesignerUtil;
+import net.sourceforge.pmd.util.fxdesigner.util.beans.SettingsOwner;
+
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ListView;
+import javafx.stage.FileChooser;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+
+public class AuxClassPathController implements Initializable, SettingsOwner {
+
+    private final DesignerRoot designerRoot;
+    private ClassLoader classLoader = getClass().getClassLoader();
+    private ValidationSupport validationSupport = new ValidationSupport();
+
+
+    @FXML
+    private Button removeFileButton;
+    @FXML
+    private Button selectFilesButton;
+    @FXML
+    private ListView<File> fileListView = new ListView<>();
+    @FXML
+    private Button moveItemUpButton;
+    @FXML
+    private Button moveItemDownButton;
+    @FXML
+    private Button setClassPathButton;
+    @FXML
+    private Button cancelButton;
+
+
+    public AuxClassPathController(ObservableList<File> auxClassPathFiles, DesignerRoot designerRoot) {
+        this.designerRoot = designerRoot;
+
+        if (auxClassPathFiles != null) {
+            fileListView.setItems(auxClassPathFiles);
```

Comment:
You can't use FXML injected fields in the contructor. Here, the ListView is not yet initialised. You have to put that in the initialise method